### PR TITLE
Fix leading zero detection

### PR DESCRIPTION
**Pull request by erpel**
While using the code from this plugin, I ran into it rejecting '0.3011.0' as invalid semver. I believe to have traced this back to the index not being taken into account in the checks for minor and patch.

This PR adresses this and adds some tests to verify the leading zero detection did not break. For these tests and test dependencies I had to turn the project into a gradle build. While I believe it does not interfere with the functionality of the jenkins library, this is not required. The relevant change could also be applied by cherry-picking the actual fix.

Kind regards and thanks for making this available.

### CODE REVIEW

**Comment by erpel** [comment]
If so inclined, one could also add these tests that I just discovered when looking at the upstream source by @onyxbits: https://raccoon.onyxbits.de/blog/single-class-java-semantic-versioning-parser-implementation/SemanticVersionTest.java

**Comment by etoews** [comment]
Thanks for the fix and the PR!

I'm not really familiar with Gradle so could you please add a "Test" section to the README.md with instructions on how to run your tests?

**Comment by etoews** [comment]
We don't want anyone executing an un-verified binary so please add the [Gradle Wrapper Validation GitHub Action](https://github.com/marketplace/actions/gradle-wrapper-validation#add-a-new-dedicated-workflow).

File: README.md [hunk 1]
```diff
@@ -1,3 +1,16 @@
 # Semantic Version Comparison in a Jenkins pipeline
 
 A Jenkins global pipeline library method [semver_compare](vars/semver_compare.groovy) and an example [Jenkinsfile](Jenkinsfile) for how to use it.
+
+## Running tests
```
Comment (etoews) [inline README.md:5]: The [rendered version of this section](https://github.com/etoews/jenkins-semver-compare/blob/4f5c601dedb3557f621bf57eef5fde5091045e82/README.md) looks a bit off.

Please use the following

```markdown
Gradle is used to run tests for the SemanticVersion class used in this library.
The repository comes with [Gradle wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html), a helper script to automatically download and manage the gradle distribution. Invoke the wrapper script according to the OS platform:

UNIX/Linux

    ./gradlew test

Windows (untested)

    gradlew.bat
```

Comment (erpel) [reply README.md:5]: Ooops, sorry for that oversight. I initially blamed the preview in IDEA, but it does not look right there either so this is purely on me.

I'd like to keep the "one line per sentence" convention, as it does not affect the output and makes future diffs easier to read - the rest of the suggested changes are applied.

**Comment by erpel** [comment]
> We don't want anyone executing an un-verified binary so please add the [Gradle Wrapper Validation GitHub Action](https://github.com/marketplace/actions/gradle-wrapper-validation#add-a-new-dedicated-workflow).

I felt a bit weird committing the jar but it did not work without it. This is my first time working with wrapper in a public repo - the Github action is a useful find - pleasure to work with you :)

**Review by etoews — APPROVED** [review]
Thanks again for the PR!

**Pull request merged**